### PR TITLE
fix(overlay): emit attach and detach at appropriate times

### LINKED ISSUE #4871: Overlay attachments observable is triggered before the attachment is complete
#### Bug, feature request, or proposal:
bug

#### What is the expected behavior?
Observable should be triggered when the attachment is complete

#### What is the current behavior?
It's triggered before that

#### What are the steps to reproduce?
see plunker

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
Plunker template: https://goo.gl/DlHd6U

http://embed.plnkr.co/aNgzEWACkFlAzOC0ZyfJ/

#### Is there anything else we should know?
push `.next()` to attachments at the end of the `attach()` method should fix the issue.

### PR DESCRIPTION
**Pull request by crisbeto**
* Emits the `attachments` when everything has been attached.
* Emits the `detachments` when everything has been detached.
* Completes the `attachments` observable before the `detachments`.

Fixes #4871.

### CODE REVIEW

File: src/lib/core/overlay/overlay-ref.ts [hunk 1]
```diff
@@ -53,6 +52,8 @@ export class OverlayRef implements PortalHost {
       this._attachBackdrop();
     }
 
+    this._attachments.next();
```
Comment (jelbourn) [inline src/lib/core/overlay/overlay-ref.ts]: Add a comment like
```ts
// Only emit the `attachments` event once all other setup is done.
```

File: src/lib/core/overlay/overlay-ref.ts [hunk 2]
```diff
@@ -68,9 +69,12 @@ export class OverlayRef implements PortalHost {
     // pointer events therefore. Depends on the position strategy and the applied pane boundaries.
     this._togglePointerEvents(false);
     this._state.scrollStrategy.disable();
+
+    let detachmentResult = this._portalHost.detach();
```
Comment (jelbourn) [inline src/lib/core/overlay/overlay-ref.ts]: Same kind of comment here

**Comment by andrewseguin** [comment]
Please rebase
